service/src/main/java/org/broadinstitute/listener/relay/Utils.java

```diff
@@ -58,7 +58,12 @@ public static boolean isValidOrigin(String origin, CorsSupportProperties corsSup
     // We want to strip the protocol.
     try {
       URL url = new URL(origin);
-      stringToCheck = url.getHost();
```

Comment:
Replacing with `url.getAuthority()` will include both host and port (if specified), as well as omitting the trailing slash.

Reply:
Oh? Let me try!